backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/validation/impl/IndexSettingsValidator.java

```diff
@@ -28,13 +28,14 @@
 	@Override
 	public void validate(ValidationErrorCollector errorCollector, IndexSettings expected, IndexSettings actual) {
 		Analysis expectedAnalysis = expected.getAnalysis();
-		if ( expectedAnalysis == null ) {
-			// No expectation
-			return;
+		if ( expectedAnalysis != null ) {
+			Analysis actualAnalysis = actual.getAnalysis();
+			validateAnalysisSettings( errorCollector, expectedAnalysis, actualAnalysis );
 		}
-		Analysis actualAnalysis = actual.getAnalysis();
 
-		validateAnalysisSettings( errorCollector, expectedAnalysis, actualAnalysis );
+		LeafValidators.EQUAL.validate( errorCollector, ValidationContextType.OTHER_INDEX_SETTINGS_ATTRIBUTE,
```

Comment:
This should use `validateWithDefault`, I think? The settings returned by Elasticsearch will not necessarily include the value of `max_result_window_size`, in which case we should assume `10,000`.

One way to test this would be to create an index manually without specifying the `max_result_window_size`, then executing validation against that index. I'm fairly sure validation would fail with this code, but it should pass.

Probably a test to add to `ElasticsearchIndexSchemaManagerValidationOtherSettingsIT`?